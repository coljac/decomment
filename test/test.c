#include <objc/runtime.h>
#include <objc/message.h>
#include <stdio.h>
/* A program to do stuff
 * What, I don't know.
 * That does nothing */

static void create_shared_application(void) {
  id nsa = objc_getClass("NSApplication");
  SEL sha = sel_registerName("sharedApplication"); // OK here
  // OK there
  id (*msg_send)(id, SEL) = (id (*)(id, SEL))objc_msgSend;
  msg_send(nsa, sha);
  printf("[macosx] hello.\r\n");
}
