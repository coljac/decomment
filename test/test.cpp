#include <iostream>
#include <filesystem>
#include <vector>
/*
 * This does nothing much
 */

class DirectoryLister {
public: // A comment here
    std::vector<std::string> listFiles(const std::string& path = ".") {
        std::vector<std::string> files;
        for (const auto& entry : std::filesystem::directory_iterator(path)) {
            files.push_back(entry.path().string());
        } /* and one there */
        return files;
    }

    void printFiles(const std::string& path = ".") {
        for (const auto& file : listFiles(path)) {
            std::cout << file << std::endl;
        }
    } // What?
};

// Nothing to see here
int main() {
    DirectoryLister lister;
    lister.printFiles();
    return 0;
}
