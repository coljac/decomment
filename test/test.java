// ListFilesInDirectory.java
import java.io.File;

// Java comment here
public class ListFilesInDirectory {
    public static void main(String[] args) {
        File folder = new File(".");
        /* This
              is
              a
              multi-line
              comment */
        File[] listOfFiles = folder.listFiles();

        if (listOfFiles != null) {
            for (File file : listOfFiles) {
                if (file.isFile()) {
                    System.out.println("File: " + file.getName());
                } else if (file.isDirectory()) {
                    System.out.println("Directory: " + file.getName());
                }
            }
        } else {
            System.out.println("Unable to list files in the directory.");
        }
    }
} // done and done

