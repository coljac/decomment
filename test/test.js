const fs = require('fs');
const path = require('path');

class DirectoryLister {
  constructor(directoryPath = '.') {
    this.directoryPath = directoryPath;
  } // Commenty comment

  listFiles() {
    fs.readdir(this.directoryPath, (err, files) => {
      if (err) {
        console.error('Error reading directory:', err);
        return;
      }
      files.forEach(file => console.log(file));
    });
  }
} // Commenty comment

/* Why are you reading this
  * You should be reading the code above
  * Not this comment
  */
const lister = new DirectoryLister();
lister.listFiles();

