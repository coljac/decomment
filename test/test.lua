-- simple_calculator.lua
function add(a, b) return a + b end

function subtract(a, b) return a - b end

function multiply(a, b) return a * b end

function divide(a, b) if b ~= 0 then return a / b else return "Error: Division by zero" end end

-- Test the simple calculator functions print("Addition: ", add(10, 5))
print("Subtraction: ", subtract(10, 5)) print("Multiplication: ",
multiply(10, 5)) print("Division: ", divide(10, 5)) print("Division by zero:
", divide(10, 0))


