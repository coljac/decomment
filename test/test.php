<?php
// comment here - is this right
class FileLister {
    public function listFiles() {
        return scandir(getcwd());
    }
}

/* Multiline 
as well
*/ 
$lister = new FileLister(); // yep
print_r($lister->listFiles()); /* nope */
?>

