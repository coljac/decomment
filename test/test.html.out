<!DOCTYPE html>
<html>
<head>
    <title>File List</title>
    <style>
        .file-list {
            font-family: Arial, sans-serif;
            margin-top: 20px;
        }
        .file-item {
            margin-bottom: 5px;
        }
    </style>
</head>
<body>
    <div id="file-list" class="file-list"></div>

    <script>
        function fetchFiles() {
            fetch('list-files')
                .then(response => response.json())
                .then(files => {
                    const fileList = document.getElementById('file-list');
                    files.forEach(file => {
                        const fileItem = document.createElement('div');
                        fileItem.className = 'file-item';
                        fileItem.textContent = file;
                        fileList.appendChild(fileItem);
                    });
                });
        }

        fetchFiles();
    </script>
</body>
</html>

const express = require('express');
const fs = require('fs');
const path = require('path');

const app = express();
const PORT = 3000;

app.use(express.static(path.join(__dirname, 'public')));

app.get('/list-files', (req, res) => {
    const directoryPath = path.join(__dirname, 'public');
    fs.readdir(directoryPath, (err, files) => {
        if (err) {
            return res.status(500).send('Unable to scan directory');
        }
        res.json(files);
    });
});

app.listen(PORT, () => {
    console.log(`Server is running on http://localhost:${PORT}`);
});

